rust/src/api.rs

```diff
@@ -168,91 +168,114 @@ impl WalletInfo {
-    stream.add(Event::Offer(offer::get_offer().await.ok()));
-
-    stream.add(Event::Init("Fetching your balance".to_string()));
-    stream.add(Event::WalletInfo(
-        WalletInfo::build_wallet_info().await.ok(),
-    ));
-    stream.add(Event::Init("Checking channel state".to_string()));
-    stream.add(Event::ChannelState(get_channel_state()));
-
-    stream.add(Event::Init("Ready".to_string()));
-    stream.add(Event::Ready);
-
-    // spawn a connection task keeping the connection with the maker alive.
-    let peer_manager = wallet::get_peer_manager()?;
-    let connection_handle = connection::spawn(peer_manager);
-
-    // sync offers every 5 seconds
-    let offer_handle = offer::spawn(stream.clone());
-
-    // sync wallet every 60 seconds
-    let wallet_sync_handle = tokio::spawn(async {
-        loop {
-            wallet::sync().unwrap_or_else(|e| tracing::error!(?e, "Failed to sync wallet"));
-            tokio::time::sleep(std::time::Duration::from_secs(60)).await;
-        }
-    });
-
-    // sync wallet info every 10 seconds
-    let wallet_info_stream = stream.clone();
-    let wallet_info_sync_handle = tokio::spawn(async move {
-        loop {
-            match WalletInfo::build_wallet_info().await {
-                Ok(wallet_info) => {
-                    let _ = wallet_info_stream.add(Event::WalletInfo(Some(wallet_info)));
+    let runtime = runtime()?;
+    runtime.block_on(async move {
+        stream.add(Event::Init(format!("Initialising {network} wallet")));
+        wallet::init_wallet(Path::new(app_dir.as_str()))?;
+
+        stream.add(Event::Init("Initialising database".to_string()));
+        db::init_db(
+            &Path::new(app_dir.as_str())
+                .join(network.to_string())
+                .join("taker.sqlite"),
+        )
+        .await?;
+
+        stream.add(Event::Init("Starting full ldk node".to_string()));
+        let background_processor = wallet::run_ldk()?;
+
+        stream.add(Event::Init("Fetching an offer".to_string()));
+        stream.add(Event::Offer(offer::get_offer().await.ok()));
+
+        stream.add(Event::Init("Fetching your balance".to_string()));
+        stream.add(Event::WalletInfo(
+            WalletInfo::build_wallet_info().await.ok(),
+        ));
+        stream.add(Event::Init("Checking channel state".to_string()));
+        stream.add(Event::ChannelState(get_channel_state()));
+
+        stream.add(Event::Init("Ready".to_string()));
+        stream.add(Event::Ready);
+
+        // spawn a connection task keeping the connection with the maker alive.
+        runtime.spawn(async move {
+            let peer_info = config::maker_peer_info();
+            loop {
+                let peer_manager = wallet::get_peer_manager();
+                connection::connect(peer_manager, peer_info).await;
+                // add a delay before retrying to connect
+                tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+            }
+        });
```

Comment:
🙃 We could extract the bodies of the spawned tasks into functions. It might aid readability.

```suggestion
       runtime.spawn(stay_connected_to_maker());
             
// ...
       
fn async stay_connected_to_maker() {
    let peer_info = config::maker_peer_info();
    loop {
        let peer_manager = wallet::get_peer_manager();
        connection::connect(peer_manager, peer_info).await;
        
        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
}       
```